src/collectives/msccl.cc

```diff
@@ -28,7 +28,7 @@ ncclResult_t mscclLoadAlgo(const char *mscclAlgoFilePath, mscclAlgoHandle_t *msc
 
   struct mscclAlgo* devAlgo;
   NCCLCHECK(ncclCudaMalloc(&devAlgo, 1));
-  CUDACHECK(hipMemcpy(devAlgo, hostAlgo, sizeof(struct mscclAlgo), hipMemcpyHostToDevice));
+  CUDACHECK(hipMemcpyAsync(devAlgo, hostAlgo, sizeof(struct mscclAlgo), hipMemcpyHostToDevice));
```

Comment:
should HIP stream be passed in here?

Reply:
@wenkaidu I’m not familiar with the logic here. I just blindly changed it to allow workloads passing graph capturing stage. Please review and advise better ways of implementation. Thanks.

Reply:
@wenkaidu @whchung If passing stream to hipMemcpyAsync, it will be captured into graph and be executed in each graph launch, causing extra overhead.

I think the proper way to solve this is to call hipMemcpyAsync in a side stream and then synchronize it, like in https://github.com/ROCmSoftwarePlatform/rccl/blob/develop/src/include/alloc.h#L184-L218. However, this is currently not allowed in ROCm.

Another temp workaround can be moving algorithm loading from mscclInternalSchedulerSelectAlgo to mscclInternalSchedulerInit, i.e. load all algorithms during mscclInit (also RCCL init). This requires RCCL init is not in graph capturing.

Reply:
Created #982 for the temp workaround.